src/app/routes/loadInitialData/index.js

```diff
@@ -11,7 +12,10 @@ const loadInitialData = async (url, routes) => {
     return {};
   }
 
-  return route.getInitialData({ match });
+  const data = await route.getInitialData({ match });
+
+  data.dials = getDials();
```

Comment:
Doesn't this mean that the dial file will be read synchronously from disk every time a request is received? Could we look at making this more efficient?

Reply:
I've made the read asynchronous so it can run in parallel with the initial data fetch

Reply:
Making it async is a good start but what I'm trying to understand is, what happens if we get 1000rps, will it try to read the dials file 1000 times, or only once? Dials file are only updated max. once/min.

Reply:
It would try to read the dials file 1000 times, but I don't understand the context at where we would be getting 1000rps. This should be handled and cached by Mozart shouldn't it? In which case it would only be requested on each article publish when it is first accessed.

This replicates the behaviour of loadInitialData which unless Mozart is caching, would also be fetch 1000 times.

Reply:
Currently AFAICS are only cached for 30s, so once they have expired Mozart will request them again. As the number of articles increases and we starting serving WS articles we may soon therefore be receiving 1000rps. Also, the data fetch is needed because the data will be different for every article, whereas the dial value will always be the same regardless of article.

Reply:
Following on from Slack, here's an issue to investigate how we can optimise this further: https://github.com/bbc/simorgh/issues/1712